### PR TITLE
add last man standing fun simulation command

### PR DESCRIPTION
**Pull request by Alexsuperfly**
### Description:

creates a last man standing command in the OSRS_Fun directory to simulate a game of last man standing in the same manner that skyra does it with its hunger games command.

This is fully adopted from their implementation and just adjusted to be themed with OSRS and some functionality was dropped since OSBot doesnt have all the utility already built in that Skyra does

### Changes:

-   [] I have tested all my changes thoroughly.

### CODE REVIEW

**Review by gc — CHANGES_REQUESTED** [review]
a few things: youre adding lms wins/points as minigames, which makes no sense, it should be just LMS, and since this is just simulation it should be stateless and have no settings changing. `LastmanStandingUsage` should be put somewhere else, its not a command. and after removing that, LMS doesnt need its own folder. it also seems like the messages are duplicated in these 2 files, shouldnt they be in only 1?

File: src/lib/LastManStandingUsage.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	}
+}
+
+export const LMS_PREP = [
+	'{1} grabs a DDS.',
+	'{1} grabs an Abyssal whip and retreats.',
+	'{1} and {2} fight for a whip. {1} gives up and retreats.',
+	'{1} and {2} fight for a whip. {2} gives up and retreats.',
+	"{1} finds a Heavy ballista, some Javelins, and an Ava's assembler.",
+	'{1} runs into upstairs in a building and hides.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:60]: '{1} runs into a building and hides upstairs.',

File: src/lib/LastManStandingUsage.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} takes a handful of Morrigan Javelins.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:61]: '{1} takes a handful of Morrigan's Javelins.',

File: src/lib/LastManStandingUsage.ts (hunk 1 above)
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:60]: '{1} runs into a building and hides upstairs.',

File: src/lib/LastManStandingUsage.ts [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+	"{1} rips a Zuriel's staff out of {2}'s hands.",
+	'{1} finds a vial full of Saradomin brew.',
+	'{1} gathers as much food as they can.',
+	'{1} grabs a Vesta longsword.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:65]: '{1} grabs a Vesta's longsword.',

File: src/lib/LastManStandingUsage.ts [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} takes a Ghrazi Rapier from inside the Trinity Outpost.',
+	'{1} finds a Crate with an Infernal cape.',
+	'{1} takes a AGS from inside the Moser Settlement.',
+	'{1}, {2}, and {3} work together to get as many supplies as possible.',
+	'{1} runs away with a pair of Eternal boots and some sharks.',
+	'{1} snatches a vial of Super restore and a Stamina potion.',
+	'{1} finds a supply box full of Granite mauls & supplies.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:72]: '{1} finds a box full of supplies and a Granite maul.',

File: src/lib/LastManStandingUsage.ts [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} searches a supply box, not realizing it is empty.',
+	'{1}, {2}, {3}, and {4} share everything they gathered before running.',
+	'{1} retrieves a Staff of the Dead from the Debtor hideout .',
+	"{1} grabs a Stamina potion while {2} gets Ahrim's robes.",
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:76]: "{1} grabs a Stamina potion while {2} finds a set of Ahrim's robes.",

File: src/lib/LastManStandingUsage.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} scares {2} away from the Moser Settlement.',
+	'{1} grabs a Dragonfire shield from the ground on The Mountain.',
+	'{1} snatches a pair of Dragon claws.',
+	'{1} grabs a lone pair of pants.',
+	'{1} trips over while running to a building, {2} picks them up, they run off together.'
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:81]: '{1} trips over while running to a building, {2} picks them up and they run off together.'

File: src/lib/LastManStandingUsage.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} forces {2} to eat pant.',
+	'{1} searches for a bloody chest.',
+	'{1} uses their bloodier key and recieves a full inventory of food.',
+	'{1K} forces {2D} to eat pant. {2D} chokes and dies.',
+	"{1K} throws a dragon knife into {2D}'s head.",
+	'{1D} begs for {2K} to kill them. They reluctantly oblige, killing {1D}.',
+	'{1K} and {2K} work together to spec out {3D}.',
+	'{1K} decapitates {2D} with an AGS spec.',
+	'{1K} sets {2D} on fire with Flames of Zamorak.',
+	"{1K} DDS specs {2D} while they're purchasing supplies.",
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:102]: "{1K} fatally DDS specs {2D} while they're purchasing supplies.",

File: src/lib/LastManStandingUsage.ts [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} trips over while running to a building, {2} picks them up, they run off together.'
+].map(string => new LastManStandingUsage(string));
+
+export const LMS_FINAL = [
+	'{1} eats a shark.',
+	'{1} attacks {2}, but they manage to escape.',
+	'{1} chases {2}.',
+	'{1} runs away from {2}.',
+	'{1} defeats {2} in a fight, but spares their life.',
+	'{1} and {2} work together for now.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:90]: '{1} and {2} work together, for now...',

File: src/lib/LastManStandingUsage.ts [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D} dies from disease.',
+	"{1K} kills {2D} with a Morrigan's throwing axe.",
+	'{1D} dies in the fog trying to escape the arena.',
+	'{1D} accidently detonates a chinchompa trying to throw it.',
+	'{1K} ambushes {2D} and kills them.',
+	'{1K} severely slices {2D} with an AGS spec.',
+	'{1K} shoots {2D} with a Dark bow.',
+	"{1K} stabs {2D} in the back with a Vesta's longsword.",
+	'{1K} kills {2D} with a Ghrazi rapier.',
+	'{1K} repeatedly stabs {2D} to death with Draggon dagger.'
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:123]: '{1D} Talks about how good they are while fighting {2K} and is promptly stacked out by {2K}',

Comment (TwiceUponATaco) [reply src/lib/LastManStandingUsage.ts:123]: as an addition

File: src/lib/LastManStandingUsage.ts [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+].map(string => new LastManStandingUsage(string));
+
+export const LMS_ROUND = [
+	'{1D} steps out of the safezone too soon and gets killed by the fog.',
+	"{1K} throws a Morrigan Javelin into {2D}'s head.",
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:128]: "{1K} throws a Morrigan's Javelin into {2D}'s head.",

File: src/lib/LastManStandingUsage.ts [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D} gets 6 Hour logged and dies.',
+	'{1K} catches {2D} off guard and AGS specs a 73.',
+	"{1K} shoots a Dragon arrow into {2D}'s head.",
+	'{1D} cannot handle the circumstances and runs in the fog.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:132]: '{1D} cannot handle the circumstances and runs off into the fog.',

File: src/lib/LastManStandingUsage.ts [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D} is unable to convince {2K} to not kill them.',
+	'{1K} convinces {2D} to not kill them, only to kill {2D} instead.',
+	'{1K}, {2}, and {3D} start fighting, but {2} runs away as {1K} kills {3D}.',
+	'{1K} kills {2D} with their own spec weapon.',
+	'{1K} overpowers {2D}, killing them.',
+	'{1K} throws a Chinchompa, killing {2D}.',
+	'{1K} throws a Chinchompa, killing {2D}, and {3D}.',
+	'{1K} throws a Chinchompa, killing {2D}, {3D}, and {4D}.',
+	'{1K} throws a Chinchompa, killing {2D}, {3D}, {4D} and {5D}.',
+	'{1K} kills {2D} as they try to run.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:153]: '{1K} kills {2D} as they try to run away.',

File: src/lib/LastManStandingUsage.ts [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D}, {2D}, {3D}, and {4D} form a suicide pact, and run in the fog.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:154]: '{1D}, {2D}, {3D}, and {4D} form a suicide pact, and run off into the fog.',

File: src/lib/LastManStandingUsage.ts [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D}, {2K}, and {3D} get into a fight. {2K} triumphantly kills them both.',
+	'{1D}, {2D}, and {3K} get into a fight. {3K} triumphantly kills them both.',
+	'{1K} finds {2D} hiding in the Debtor Hideout and kills them.',
+	'{1D} finds {2K} hiding in the Debtor Hideout, but {2K} kills them.',
+	'{1K} kills {2D} with a Dragon scimitar.',
+	'{1K} and {2D} fight for a crate.{1K} strangles {2D} with a whip and runs.',
+	'{1D} and {2K} fight for a crate.{2K} strangles {1D} with a whip and runs.',
+	'{1K} repeatedly stabs {2D} to death with a Ghrazi Rapier.',
+	'{1D} trips over while running from the fog, and is killed by {2K}.',
+	'{1} trips over while running from the fog, {2} picks them up, they run off together.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:176]: '{1} trips over while running from the fog, {2} picks them up and they run off together.',

File: src/lib/LastManStandingUsage.ts [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+	'{1} forces {2} to eat pant.',
+	'{1K} forces {2D} to eat pant. {2D} chokes and dies.',
+	'{1K} catches {2D} off guard and kills them.',
+	"{1K} throws a dragon knife into {2D}'s head.",
+	'{1D} begs for {2K} to kill them. They reluctantly oblige, killing {1D}.',
+	'{1K} and {2K} work together to kill {3D}.',
+	"{1K} shoots a Dark bow spec into {2D}'s head.",
+	'{1D} bleeds out due to untreated injuries.',
+	'{1D} cannot handle the circumstances and logs out.',
+	'{1D} unknowingly eats cavada berries.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:201]: '{1D} unknowingly eats cavada berries and dies.',

File: src/lib/LastManStandingUsage.ts [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+	'{1K} convinces {2D} to not kill them, only to kill {2D} instead.',
+	'{1K}, {2}, and {3D} start fighting, but {2} runs away as {1K} kills {3D}.',
+	'{1K} kills {2D} with their own weapon.',
+	'{1K} overpowers {2D}, killing them.',
+	'{1K} kills {2D} as they try to run.',
+	'{1K} kills {2D} with a dragon hatchet.',
+	'{1K} and {2K} fight {3D} and {4D}. {1K} and {2K} survive.',
+	'{1D} and {2D} fight {3K} and {4K}. {3K} and {4K} survive.',
+	'{1D} dies in the fog trying to escape the arena.',
+	'{1D} accidently detonates a land mine while trying to arm it.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:222]: '{1D} accidentally detonates a Chinchompa while trying to pet it.',

Comment (TwiceUponATaco) [reply src/lib/LastManStandingUsage.ts:222]: There are no land mines in runescape :)

File: src/lib/LastManStandingUsage.ts [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+	'{1D} attacks {2}, but {3K} protects them, killing {1D}.',
+	'{1K} ambushes {2D} and kills them.',
+	'{1D} accidently sets off a wild chinchompa, getting blown to pieces.',
+	"{1K} severely slices {2D} with a Vesta's longsword.",
+	'{1K} strangles {2D} with an Abyssal whip.',
+	'{1K} kills {2D} for their supplies.',
+	'{1K} shoots an arrow at {2}, but misses and kills {3D} instead.',
+	"{1K} shoots a poisonous blow dart into {2D}'s neck, slowly killing them.",
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:230]: "{1K} shoots a poisonous bronze dart into {2D}'s neck, slowly killing them.",

**Comment by osrsolympus** [comment]
Perhaps a “**User** is found to be using an auto-prayer client and is banned mid game“ line for the lols

File: src/lib/LastManStandingUsage.ts [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+	"{1K} aims a Dragon arrow at {2}'s head and shoots, {3D} jumps in the way and sacrifices their life to save them.",
+	'{1} searches for a bloody chest.',
+	'{1} hides in a building to rest.',
+	'{1} eats all their food.',
+	'{1} and {2} run into each other and decide to truce for the round.',
+	'{1} thinks about winning.',
+	'{1} looks at the night sky.',
+	'{1} defeats {2} in a fight, but spares their life.',
+	'{1} begs for {2} to kill them. They refuse, keeping {1} alive.',
+	'{1} receives a Dragon hatchet from an unknown sponsor.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:187]: '{1} receives a Dragon hatchet from a random chest.',

File: src/lib/LastManStandingUsage.ts [hunk 19]
```diff
@@ -0,0 +1,247 @@
...
+	'{1} receives a combat potion from an unknown sponsor.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:188]: '{1} receives a combat potion from a random chest.',

File: src/lib/LastManStandingUsage.ts [hunk 20]
```diff
@@ -0,0 +1,247 @@
...
+	'{1} receives a bag of potions from an unknown sponsor.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:189]: '{1} receives a bag of potions from a random chest.',

File: src/lib/LastManStandingUsage.ts [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+	'{1} receives food from an unknown sponsor.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:190]: '{1} receives food from a random chest.',

File: src/lib/LastManStandingUsage.ts [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+	'{1} receives a chinchompa from an unknown sponsor.',
```
Comment (TwiceUponATaco) [inline src/lib/LastManStandingUsage.ts:191]: '{1} receives a chinchompa from a random chest.',

**Review by TwiceUponATaco — CHANGES_REQUESTED** [review]
Changed unknown sponsor to random chest to fit the LMS theme better

**Comment by TwiceUponATaco** [comment]
Suggested Lines:
1 trips over a chinchompa while running away from 2, blowing themself up.
1 trips over a chinchompa while running away from 2, blowing them both up.

**Pull request merged**